sklearnex/ensemble/forest.py

```diff
@@ -13,6 +14,990 @@
+            'variable_importance_mode' : 'mdi'
+        }
+        self._cached_estimators_ = None
+        self._onedal_estimator = onedal_RandomForestRegressor(**onedal_params)
+        self._onedal_estimator.fit(X, y, sample_weight, queue=queue)
+
+        self._save_attributes()
+        if sklearn_check_version("1.2"):
+            self._estimator = DecisionTreeRegressor()
+        self.estimators_ = self._estimators_
+        return self
+
+    def _onedal_predict(self, X, queue=None):
+        if sklearn_check_version("1.0"):
+            self._check_feature_names(X, reset=False)
+        X = check_array(
+            X,
+            accept_sparse=False,
+            dtype=[np.float64, np.float32]
+        )
+        return self._onedal_estimator.predict(X, queue=queue)
+
+    @wrap_output_data
+    def fit(self, X, y, sample_weight=None):
+        """
+        Build a forest of trees from the training set (X, y).
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            The training input samples. Internally, its dtype will be converted
+            to ``dtype=np.float32``. If a sparse matrix is provided, it will be
+            converted into a sparse ``csc_matrix``.
+
+        y : array-like of shape (n_samples,) or (n_samples, n_outputs)
+            The target values (class labels in classification, real numbers in
+            regression).
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights. If None, then samples are equally weighted. Splits
+            that would create child nodes with net zero or negative weight are
+            ignored while searching for a split in each node. In the case of
+            classification, splits are also ignored if they would result in any
+            single class carrying a negative weight in either child node.
+
+        Returns
+        -------
+        self : object
+        """
+        if not self.bootstrap and self.max_samples is not None:
+            raise ValueError(
+                "`max_sample` cannot be set if `bootstrap=False`. "
+                "Either switch to `bootstrap=True` or set "
+                "`max_sample=None`."
+            )
+        # We have to get `n_outputs_` before dispatching
+        # oneDAL requirements: Number of outputs `n_outputs_` should be 1.
+        y = np.asarray(y)
+
+        if y.ndim == 2 and y.shape[1] == 1:
+            warnings.warn("A column-vector y was passed when a 1d array was"
+                          " expected. Please change the shape of y to "
+                          "(n_samples,), for example using ravel().",
+                          DataConversionWarning, stacklevel=2)
+
+        if y.ndim == 1:
+            # reshape is necessary to preserve the data contiguity against vs
+            # [:, np.newaxis] that does not.
+            y = np.reshape(y, (-1, 1))

```

Comment:
rephrase comment?

Reply:
code styling and cleaning will be done :)

Reply:
> rephrase comment?

This comment is ok.

Reply:
I understand, just wanted to keep a pointer so that it becomes easier to find later.